### DISCUSSION TITLE
Orgin Validation

### DISCUSSION THREAD

**Discussion by santhoshit8** [webauthn4j/webauthn4j#708] [General]
Hi All,

If sub domain is allowed, do we need to pass all possible subdomain orgin in orgin list ?

Is there any other way?

**Comment by ynojima** [comment]
If you want to allow a specific subdomain, set expected FQDN to `ServerProperty`'s`origin`.
https://github.com/webauthn4j/webauthn4j/blob/master/webauthn4j-core/src/main/java/com/webauthn4j/server/ServerProperty.java#L51

If you want to allow multiple subdomains, set expected FQDNs to `ServerProperty`'s`origins`.
https://github.com/webauthn4j/webauthn4j/blob/master/webauthn4j-core/src/main/java/com/webauthn4j/server/ServerProperty.java#L64

If you want to allow muptiple subdomains, but which are hard to be listed, implement custom `OriginValidator` as you like, and set it to `RegistrationDataValidator` like this:
https://github.com/webauthn4j/webauthn4j/blob/3d96439b25d790baf9d6156e447ea20089a86090/webauthn4j-core/src/test/java/integration/component/CustomOriginValidatorTest.java#L64

**Reply by santhoshit8** [reply]
Thanks for info @ynojima.
